### PR TITLE
allowing the ability to show joke on screen

### PR DESCRIPTION
**Pull request by chenry**
While this does not show the random image, it is getting a step closer by showing a joke from the API.

had to call the function on the main thread in order to
reflect the changes in the UI.

https://github.com/GRCocoaHeads/ChuckNorrisIOS/issues/29

I think that this pull request closes this issue as well:
https://github.com/GRCocoaHeads/ChuckNorrisIOS/issues/14

### CODE REVIEW

File: CHChuckNorris/CHChuckNorris/AppDelegate.swift [hunk 1]
```diff
@@ -12,10 +12,11 @@ import UIKit
 class AppDelegate: UIResponder, UIApplicationDelegate {
                             
     var window: UIWindow?
-
+    var jokeController: JokeController?
```
Comment (manlycode) [inline CHChuckNorris/CHChuckNorris/AppDelegate.swift:15]: Use a "let" here. That will give a warning if the jokeController variable gets reassigned.

Comment (chenry) [inline CHChuckNorris/CHChuckNorris/AppDelegate.swift:15]: @manlycode I attempted to change it to a 'let' but unfortunately I am getting compilation errors that I do not know how to resolve.  Perhaps you can point me in the right direction....

"Cannot assign to 'jokeController' in 'self'"

Comment (manlycode) [inline CHChuckNorris/CHChuckNorris/AppDelegate.swift:15]: My mistake. I made that comment off-the-cuff without hand-holding from the IDE.

**Comment by manlycode** [comment]
The code looks good. I merged this pulled the changes, and then ran it, and noticed a design bug:

After the joke loads, I'm able to edit the joke.

screen shot 2014-10-02 at 11 10 58 am

**Comment by mccrackend** [comment]
@tbugai Is that a common design pattern for iOS? to have object builder logic inside the model classes themselves, along with code calling out to external systems to fetch the data?  In the java world, we typically have dumb models, controllers that.. control, and dao's to fetch data from external systems. It's supposed to help keep things orderly for http://en.wikipedia.org/wiki/Separation_of_concerns reasons

**Comment by tbugai** [comment]
It is a common design pattern in iOS, Ruby, and a few others.  Separation of concerns is a great idea in theory but having the logic broken out into so many different places makes it harder to follow and thus harder to maintain.  

Keeping the Joke class only responsible for jokes (whether it be fetching them from the server or just being a plain old object) should keep it small enough to maintain properly.

**Pull request merged**